packages/tfchain_client/src/client.ts

```diff
@@ -294,11 +300,11 @@ class Client extends QueryClient {
           // seed shouldn't have spaces
           throw new ValidationError("Invalid mnemonic! Must be bip39 compliant");
 
-        if (!this.mnemonicOrSecret.startsWith("0x"))
-          throw new ValidationError("Invalid secret seed. secret seed should starts with 0x");
-        const secret = this.mnemonicOrSecret.substring(2);
-        if (secret.length !== 64) throw new ValidationError("Invalid secret length. Secret length should be 64");
-        if (!isValidSeed(secret)) throw new ValidationError("Invalid secret seed");
+        if (this.mnemonicOrSecret.startsWith("0x")) {
```

Comment:
too many nested level here, I suggest you change the checkInputs to be something like

```
function checkInputs(): void {
  // Validate URL presence
  if (!this.url) {
    throw new ValidationError("URL is required.");
  }

  // Validate Keypair Type
  if (!SUPPORTED_KEYPAIR_TYPES.includes(this.keypairType)) {
    const validTypes = SUPPORTED_KEYPAIR_TYPES.join(", ");
    throw new ValidationError(
      `Invalid keypair type: "${this.keypairType}". Valid options are: ${validTypes}.`
    );
  }

  // Ensure either mnemonicOrSecret or extSigner is provided
  if (!this.mnemonicOrSecret && !this.extSigner) {
    throw new ValidationError("Either 'mnemonicOrSecret' or 'extSigner' must be provided.");
  }

  // Early exit if mnemonicOrSecret is not provided
  if (!this.mnemonicOrSecret) {
    return;
  }

  // Early exit if mnemonicOrSecret indicates an extension signer
  if (this.mnemonicOrSecret.startsWith("//")) {
    return;
  }

  // Process secret by removing '0x' prefix if present
  const secret = this.mnemonicOrSecret.startsWith("0x")
    ? this.mnemonicOrSecret.slice(2)
    : this.mnemonicOrSecret;

  // Validate mnemonic
  if (!validateMnemonic(this.mnemonicOrSecret)) {
    throw new ValidationError("Invalid mnemonic provided.");
  }

  // Validate secret length
  if (secret.length !== 64) {
    throw new ValidationError("Secret must be exactly 64 characters long.");
  }

  // Validate seed derived from the secret
  if (!isValidSeed(secret)) {
    throw new ValidationError("Invalid seed derived from the secret.");
  }
}

```

So you can make use of early exits

Reply:
done